hashicorp/boundary-ui
# Icu 12782 add filtering pagination support for credential stores

✅ Closes: https://hashicorp.atlassian.net/browse/ICU-12782

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12782)

## Description


Add search, filtering, and pagination support for credential stores. User can search by name, description, id and type.

NOTE: This PR adds a change to the indexedDB and this will result in errors if you have a diff indexedDB schema loaded already. To bypass this delete the indexedDB by going to devtools -> application -> IndexedDB in order to get the changes this PR introduces.

## Screenshots (if appropriate):

https://github.com/hashicorp/boundary-ui/assets/107949262/474c9f0e-4d5b-4ba6-b191-85fd72115a3c


## How to Test


1. start a boundary dev instance
2. test that search & filtering work as expected
3. validate that deleting a credential store creates no errors
4. validate that credential stores populated the correct scopes.
5. validate that the correct credentials appear in the brokered/injected credentials tabs for targets


:technologist: [Admin preview](PATH_TO_FEATURE)


## Checklist:
